lib/srv/app/server.go

```diff
@@ -228,12 +242,15 @@ func New(ctx context.Context, c *Config) (*Server, error) {
 		dynamicLabels: make(map[string]*labels.Dynamic),
 		apps:          make(map[string]types.Application),
 		awsSigner:     awsSigner,
+		connLimiter:   c.ConnectionLimiter.ConnectionsLimiter,
```

Comment:
I'm not sure this extra field is necessary TBH. You can just use `ConnectionLimiter` from config.